### ISSUE TITLE
🐛 should report duplicate identifier error

### ISSUE THREAD

**Issue by bvanjoi** [biomejs/biome#2394]
### Environment information

```block
CLI:
  Version:                      1.6.4
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v20.11.1"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "npm/10.2.4"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Workspace:
  Open Documents:               0
```

### What happened?

- code:

```ts
// npx @biomejs/biome check --apply code.ts
function f(a: number) {
  const a = 1; // expected throw error here but not.
}
```

- playground:

https://biomejs.dev/playground/?code=ZgB1AG4AYwB0AGkAbwBuACAAZgAoAGEAOgAgAG4AdQBtAGIAZQByACkAIAB7AAoAIAAgAGMAbwBuAHMAdAAgAGEAIAA9ACAAMQA7AAoAfQA%3D

### Expected result

Report errors like ﻿`duplicated identifier`

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by 10Derozan** [comment]
Similarly, type parameters should not be in different scopes too.
For example:
```
function identity<T>(arg: T): T {
    const a: T = 1; // expect error
    return arg;
}
```

**Comment by Conaclos** [comment]
> function identity<T>(arg: T): T {
>     const a: T = 1; // expect error
>     return arg;
> }

I don't see any issue in your example...

**Comment by 10Derozan** [comment]
> > function identity(arg: T): T {
> > const a: T = 1; // expect error
> > return arg;
> > }
> 
> I don't see any issue in your example...

Sorry, maybe they are not the similar error, in ts check it will fail.

The example should be 
```
function identity<T>(arg: T): T {
    type T = number;
}
```
